### PR TITLE
Rewrite messages reducer and actions with RTK

### PR DESCRIPTION
**Pull request by markerikson**
This PR:

- Removes the apparently-unused "message payloads" field and logic
- Rewrites the `messages` reducer to use RTK's `createSlice`, and alters the `state.messages` structure:
    - Changed `messages` and `logpointMessages` from `Map` instances to RTK `EntityState` lookup tables
    - Updated the `addMessage/removeMessages` functions to truly "mutate" the state, thanks to RTK and Immer
    - Changed `removedLogpointIds` from a `Set` to an array
    - Changed `map.get(id)` calls to `.entities[id]`
    - Added `getTotalMessagesCount`
- Reworks `ConsoleOutput` to simplify `lastMessage` calculation
- Migrates `actions/messages` to TS
- Replaces all existing usages of the messages action creators with the slice-generated ones
- Disables text search of "grip preview" items to work around a circular import issue I was seeing

**Status**

This is not 100% complete yet. I've left the dead reducer code in there for comparison, as I'm still doing hands-on checking and confirmation of behavior.  But, so far I can see that messages are being added and removed, evaluations and logpoints are clearing, and I can toggle messages open and closed. That looks encouraging.

Will come back to this tomorrow to do final checks and cleanup, but I think the primary changes are done.

### CODE REVIEW

**Comment by markerikson** [comment]
Some hands-on smoke testing looks good atm. Going to clean up the dead code now.

**Comment by markerikson** [comment]
Cleanup pushed. Just waiting for a review stamp!

File: src/devtools/client/webconsole/components/Output/ConsoleOutput.tsx [hunk 1]
```diff
@@ -88,12 +89,22 @@ class ConsoleOutput extends React.Component<PropsFromRedux> {
...
+
+    // We're not sorting the IDs array, so should be latest added
+    const lastMessageId = ids[ids.length - 1];
+    const lastMessage = entities[lastMessageId]!;
 
     // [...this.props.messages.values()] seems slow
     // we should have a separate messageList somewhere we can check OR
     // use a memoization function to be able to get the last message quickly
-    const lastMessage = [...this.props.messages.values()][this.props.messages.size - 1];
+    // const lastMessage = [...this.props.messages.values()][this.props.messages.size - 1];
```
Comment (jaril) [inline src/devtools/client/webconsole/components/Output/ConsoleOutput.tsx:106]: Can we delete this?

**Review by jaril — APPROVED** [review]

File: src/devtools/client/webconsole/components/Output/ConsoleOutput.tsx (hunk 1 above)
Comment (markerikson) [reply src/devtools/client/webconsole/components/Output/ConsoleOutput.tsx:106]: Ah yes, that should go away.  I'll take care of that in the follow-up PR that I'm already working on.

File: src/devtools/client/webconsole/actions/messages.ts [hunk 2]
```diff
@@ -41,16 +42,16 @@ export function setupMessages(store) {
   ThreadFront.findConsoleMessages(
     (_, msg) => store.dispatch(onConsoleMessage(msg)),
     () => store.dispatch(onConsoleOverflow())
-  ).then(() => store.dispatch({ type: "MESSAGES_LOADED" }));
+  ).then(() => store.dispatch(messagesLoaded()));
 }
 
-function convertStack(stack, { frames }) {
+function convertStack(stack: string[], { frames }: { frames?: Frame[] }) {
```
[the remark this answers is not in the document]
Comment (markerikson) [reply src/devtools/client/webconsole/actions/messages.ts:48]: The types here were based on what's already being passed in, and it seems like `frames` is typed as being optional.

**Pull request merged**